### ISSUE TITLE
from_flax only working with safety_checker=None

### ISSUE THREAD

**Issue by camenduru** [huggingface/diffusers#1987]
### Describe the bug

when I try converting without `safety_checker=None` 
`loaded_sub_model` pointing to transformers/modeling_flax_pytorch_utils.py 😐

https://github.com/huggingface/diffusers/blob/135567f18edc0bf02d515d5c76cc736d1ebddad3/src/diffusers/pipelines/pipeline_utils.py#L735

### Reproduction

working
`pipe = StableDiffusionPipeline.from_pretrained("flax", safety_checker=None, from_flax=True)`

not working sorry 😔
`pipe = StableDiffusionPipeline.from_pretrained("flax", from_flax=True)`

### Logs

```shell
File "app.py", line 89, in flax_to_pt
  pipe = StableDiffusionPipeline.from_pretrained("flax", from_flax=True)
File "/home/user/.local/lib/python3.8/site-packages/diffusers/pipelines/pipeline_utils.py", line 735, in from_pretrained
  loaded_sub_model = load_method(os.path.join(cached_folder, name), **loading_kwargs)
File "/home/user/.local/lib/python3.8/site-packages/transformers/modeling_utils.py", line 2358, in from_pretrained
  model = load_flax_checkpoint_in_pytorch_model(model, resolved_archive_file)
File "/home/user/.local/lib/python3.8/site-packages/transformers/modeling_flax_pytorch_utils.py", line 230, in load_flax_checkpoint_in_pytorch_model
  flax_cls = getattr(transformers, "Flax" + model.__class__.__name__)
File "/home/user/.local/lib/python3.8/site-packages/transformers/utils/import_utils.py", line 1086, in __getattr__
  raise AttributeError(f"module {self.__name__} has no attribute {name}")
AttributeError: module transformers has no attribute FlaxStableDiffusionSafetyChecker
```

### System Info

https://huggingface.co/spaces/camenduru/inspector

**Comment by patrickvonplaten** [comment]
Ah yes True! Thanks for the issue.

Think there are two solutions:

1. We add a hack to: https://github.com/huggingface/diffusers/blob/135567f18edc0bf02d515d5c76cc736d1ebddad3/src/diffusers/pipelines/pipeline_utils.py#L289 
2. We just add an error message if `safety_checker != None` is passed that states that the safety checker should be loaded separately.

Think I'd be in favor of 2. actually . @pcuenca what do you think?

**Comment by pcuenca** [comment]
I think I prefer `2` as well, will take a closer look tomorrow.

**Comment by pcuenca** [comment]
Fixed in #2187
